consensus/src/epoch_manager.rs

```diff
@@ -275,6 +294,98 @@ impl EpochManager {
         Ok(())
     }
 
+    fn prepare_decoupled_execution(
+        &mut self,
+        epoch: u64,
+        recovery_data: RecoveryData,
+        epoch_state: &EpochState,
+        safety_rules_container: Arc<Mutex<MetricsSafetyRules>>,
+        network_sender: &NetworkSender,
+    ) -> (
```

Comment:
this should return (RoundManager, ExecutionPhase, CommitPhase)

Reply:
resolved